### PR TITLE
Trying to make sure there is no active tasks and introduce a timeout

### LINKED ISSUE #34432: Flaky test: org.keycloak.testsuite.broker.KcSamlBrokerTest#testPostBrokerLoginFlowWithOTP_bruteForceEnabled
## org.keycloak.testsuite.broker.KcSamlBrokerTest#testPostBrokerLoginFlowWithOTP_bruteForceEnabled
[Keycloak CI - FIPS IT (non-strict)](https://github.com/keycloak/keycloak/actions/runs/11564854837/job/32191018577)

```
java.lang.AssertionError: expected:<3> but was:<2>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
...
```

### LINKED ISSUE #34474: Flaky test: org.keycloak.testsuite.broker.KcOidcBrokerTest#testPostBrokerLoginFlowWithOTP_bruteForceEnabled
## org.keycloak.testsuite.broker.KcOidcBrokerTest#testPostBrokerLoginFlowWithOTP_bruteForceEnabled
[Keycloak CI - Java Distribution IT (windows-latest - temurin - 17)]()

```
java.lang.AssertionError: expected:<3> but was:<2>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
...
```

### PR DESCRIPTION
**Pull request by pedroigor**
Closes #34432
Closes #34474

### CODE REVIEW

**Comment by pedroigor** [comment]
@douglaspalmer Pretty much what you did but relying on the active task count plus a timeout. Let's see again how it goes ...

**Review by douglaspalmer — APPROVED** [review]
Looks good; my only concern is that getActiveCount() returns an approximate number of active threads. But the extra timeout should alleviate that.

File: testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/util/WaitUtils.java [hunk 1]
```diff
@@ -162,28 +165,28 @@ public static void waitForModalFadeOut() {
...
-            assertEquals(numExecutors, threadPoolExecutor.getCompletedTaskCount());
+            try {
+                CompletableFuture.runAsync(() -> {
+                    do {
+                        try {
+                            Thread.sleep(1000);
+                        } catch (InterruptedException e) {
+                            throw new RuntimeException(e);
+                        }
+                    } while (threadPoolExecutor.getActiveCount() > 0);
```
Comment (douglaspalmer) [inline testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/util/WaitUtils.java:181]: I think we should check !threadPoolExecutor.getQueue().isEmpty() AND threadPoolExecutor.getActiveCount() > 0 here. IIRC, executors that are in the queue are not active. So this loop could theoretically terminate while executors are still queued.

**Review by douglaspalmer — APPROVED** [review]
LGTM

**Review by ahus1 — APPROVED** [review]
Approving based on previous review

**Pull request merged**